### PR TITLE
Redis based cache

### PR DESCRIPTION
**Pull request by ilia-tsyplenkov**
Redis based cache. In general it has same API as built-in dict

### CODE REVIEW

File: mindsdb/utilities/cache.py [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+import json
 
+import walrus
 from mindsdb.utilities.config import Config
 
+CONFIG = Config()
+
 
-class Cache:
+class LocalCache:
```
Comment (George3d6) [inline mindsdb/utilities/cache.py:11]: There should be a common interface between this and `RedisCache`

File: mindsdb/utilities/cache.py [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
     def __init__(self, name, *args, **kwargs):
         self.kwargs = kwargs
-        self.cache_file = os.path.join(Config()['paths']['cache'], name)
+        self.cache_file = os.path.join(CONFIG['paths']['cache'], name)
```
Comment (George3d6) [inline mindsdb/utilities/cache.py:14]: Create this config object on `__init__` on `self.config`, similar to the pattern in all other interfaces. It's very cheap to create, don't worry.

File: mindsdb/utilities/cache.py [hunk 3]
```diff
@@ -40,3 +44,54 @@ def delete(self):
         except Exception:
             pass
         os.remove(self.cache_file)
+
+
+class RedisCache:
+    def __init__(self, prefix, *args, **kwargs):
+        self.prefix = prefix
+        if CONFIG["cache"]["type"] != "redis":
```
Comment (George3d6) [inline mindsdb/utilities/cache.py:52]: same thing, create this in init

**Review by George3d6 — CHANGES_REQUESTED** [review]

File: mindsdb/utilities/cache.py (hunk 1 above)
Comment (ilia-tsyplenkov) [reply mindsdb/utilities/cache.py]: Done

File: mindsdb/utilities/cache.py (hunk 3 above)
Comment (ilia-tsyplenkov) [reply mindsdb/utilities/cache.py]: Done

File: mindsdb/utilities/cache.py (hunk 2 above)
Comment (ilia-tsyplenkov) [reply mindsdb/utilities/cache.py]: Done

**Pull request merged**